trie-db/src/nibble/nibblevec.rs

```diff
@@ -231,6 +233,39 @@ impl NibbleVec {
 		}
 		true
 	}
+
+	/// Return an iterator over `Partial` bytes representation.
+	pub fn right_iter<'a>(&'a self) -> impl Iterator<Item = u8> + 'a {
+		let require_padding = self.len % nibble_ops::NIBBLE_PER_BYTE != 0;
+		let mut ix = 0;
+		let inner = &self.inner;
+
+		crate::rstd::iter::from_fn(move || {
+			if require_padding && ix < inner.len() {
+				if ix == 0 {
+					ix += 1;
+					Some(inner[ix - 1] >> nibble_ops::BIT_PER_NIBBLE)
+				} else {
+					ix += 1;
+
+					let first_nibble = nibble_ops::pad_right(inner[ix - 2]);
+					let second_nibble = nibble_ops::at_left(0, inner[ix - 1]);
+
+					Some(nibble_ops::push_at_left(
+						0,
+						first_nibble,
+						nibble_ops::push_at_left(1, second_nibble, 0),
+					))
```

Comment:
```suggestion
					Some(inner[ix - 2] << s1 | inner[ix - 1] >> s2)
```

with `		let (s1, s2) = nibble_ops::SPLIT_SHIFTS;` outside `from_fn` is an alternate implementation.